python-package/lightgbm/engine.py

```diff
@@ -370,6 +370,8 @@ def cv(params, train_set, num_boost_round=10,
 
     if metrics:
         params.setdefault('metric', [])
+        if isinstance(params['metric'], string_type):
```

Comment:
OK，thanks! I get it. If you want to change the default loss function no matter what it is  you must write your own code. Then I agree with you to override and specify it in docs .  I will resubmit pull request you can choose one of them or fix it by yourself.